// devextreme-vue/esm/load-indicator.js
/*!
 * devextreme-vue
 * Version: 25.1.3
 * Build date: Wed Jun 25 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

import { defineComponent } from "vue";
import { prepareComponentConfig } from "./core/index";
import LoadIndicator from "devextreme/ui/load_indicator";
const componentConfig = {
    props: {
        elementAttr: Object,
        height: [Number, String],
        hint: String,
        indicatorSrc: String,
        onContentReady: Function,
        onDisposing: Function,
        onInitialized: Function,
        onOptionChanged: Function,
        rtlEnabled: Boolean,
        visible: Boolean,
        width: [Number, String]
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:elementAttr": null,
        "update:height": null,
        "update:hint": null,
        "update:indicatorSrc": null,
        "update:onContentReady": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onOptionChanged": null,
        "update:rtlEnabled": null,
        "update:visible": null,
        "update:width": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = LoadIndicator;
        this.$_hasAsyncTemplate = true;
    }
};
prepareComponentConfig(componentConfig);
const DxLoadIndicator = defineComponent(componentConfig);
export default DxLoadIndicator;
export { DxLoadIndicator };
