// devextreme-vue/cjs/stepper.js
/*!
 * devextreme-vue
 * Version: 25.1.3
 * Build date: Wed Jun 25 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DxItem = exports.DxStepper = void 0;
const vue_1 = require("vue");
const index_1 = require("./core/index");
const stepper_1 = __importDefault(require("devextreme/ui/stepper"));
const index_2 = require("./core/index");
const componentConfig = {
    props: {
        accessKey: String,
        activeStateEnabled: Boolean,
        dataSource: [Array, Object, String],
        disabled: Boolean,
        elementAttr: Object,
        focusStateEnabled: Boolean,
        height: [Number, String],
        hint: String,
        hoverStateEnabled: Boolean,
        items: Array,
        itemTemplate: {},
        linear: Boolean,
        onDisposing: Function,
        onInitialized: Function,
        onItemClick: Function,
        onItemContextMenu: Function,
        onItemRendered: Function,
        onOptionChanged: Function,
        onSelectionChanged: Function,
        onSelectionChanging: Function,
        orientation: String,
        rtlEnabled: Boolean,
        selectedIndex: Number,
        selectedItem: {},
        selectOnFocus: Boolean,
        tabIndex: Number,
        visible: Boolean,
        width: [Number, String]
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:accessKey": null,
        "update:activeStateEnabled": null,
        "update:dataSource": null,
        "update:disabled": null,
        "update:elementAttr": null,
        "update:focusStateEnabled": null,
        "update:height": null,
        "update:hint": null,
        "update:hoverStateEnabled": null,
        "update:items": null,
        "update:itemTemplate": null,
        "update:linear": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onItemClick": null,
        "update:onItemContextMenu": null,
        "update:onItemRendered": null,
        "update:onOptionChanged": null,
        "update:onSelectionChanged": null,
        "update:onSelectionChanging": null,
        "update:orientation": null,
        "update:rtlEnabled": null,
        "update:selectedIndex": null,
        "update:selectedItem": null,
        "update:selectOnFocus": null,
        "update:tabIndex": null,
        "update:visible": null,
        "update:width": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = stepper_1.default;
        this.$_hasAsyncTemplate = true;
        this.$_expectedChildren = {
            item: { isCollectionItem: true, optionName: "items" }
        };
    }
};
(0, index_1.prepareComponentConfig)(componentConfig);
const DxStepper = (0, vue_1.defineComponent)(componentConfig);
exports.DxStepper = DxStepper;
const DxItemConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:disabled": null,
        "update:hint": null,
        "update:icon": null,
        "update:isValid": null,
        "update:label": null,
        "update:optional": null,
        "update:template": null,
        "update:text": null,
    },
    props: {
        disabled: Boolean,
        hint: String,
        icon: String,
        isValid: Boolean,
        label: String,
        optional: Boolean,
        template: {},
        text: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxItemConfig);
const DxItem = (0, vue_1.defineComponent)(DxItemConfig);
exports.DxItem = DxItem;
DxItem.$_optionName = "items";
DxItem.$_isCollectionItem = true;
exports.default = DxStepper;
